cirq-core/cirq/protocols/qasm.py

```diff
@@ -55,14 +55,27 @@ def __init__(
         self.qubit_id_map = {} if qubit_id_map is None else qubit_id_map
         self.meas_key_id_map = {} if meas_key_id_map is None else meas_key_id_map
 
+    def _format_number(self, value) -> str:
+        """OpenQASM 2.0 does not support '1e-5' and wants '1.0e-5'"""
+        return (
+            f'{value}'
+            if self.version != '2.0'
+            or isinstance(value, int)
+            or abs(value) > 1e-5
+            or self.precision <= 5
+            else f'{value:.{self.precision-5}E}'
+        )
```

Comment:
Perhaps we could unconditionally format with a decimal digit before `e` which is OK with either OpenQASM version.
That way the test cases would be the same regardless of the default `QasmArgs.version`.

```suggestion
        s = f'{value}'
        if 'e' in s and not '.' in s:
            s = s.replace('e', '.0e')
```

Reply:
SGTM, will commit and fix tests.